### ISSUE TITLE
[TF2] Variable List Shape

### ISSUE THREAD

**Issue by alanhdu** [tensorflow/tensorflow#29178]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): `pip install tf-nightly-2.0-preview`
- TensorFlow version (use command below): `v1.12.1-2995-ge09b015 2.0.0-dev20190530`
- Python version: 3.8

**Describe the current behavior**
Right now, the `tf.Variable` constructor accepts a list as a shape -- but if you do this, then calls to `.assign` raise an error. 

**Describe the expected behavior**
Either `tf.Variable` should automatically convert the list to a `tf.TensorShape` (or it could raise a type error for a list shape, but that seems significantly less ergonomic).

**Code to reproduce the issue**
The following code:

```python
import tensorflow as tf

x = tf.Variable(tf.zeros([50, 50]), shape=[None, 50])
x.assign(tf.zeros([10, 50]))
```

fails with:
```python
Traceback (most recent call last):
  File "a.py", line 4, in <module>
    x.assign(tf.zeros([10, 50]))
  File "/opt/anaconda/envs/tf2/lib/python3.7/site-packages/tensorflow/python/ops/resource_variable_ops.py", line 1145, in assign
    self._shape.assert_is_compatible_with(value_tensor.shape)
AttributeError: 'list' object has no attribute 'assert_is_compatible_with'
```

If I use `shape=tf.TensorShape([None, 50])` then this works just fine.

**Comment by achandraa** [comment]
Have tried with TensorFlow version 2.0.0-dev20190531 on Colab and was able to reproduce the issue.

**Comment by PiyushDatta** [comment]
@alanhdu Hi, I've tried this on tensorflow 2.1 and tf-nightly and haven't got this issue. Can you please check on your end?

**Comment by gadagashwini** [comment]
@alanhdu, This is fixed in Tf-nightly==2.2.0.dev20200312.
Please find the gist [here](https://colab.sandbox.google.com/gist/gadagashwini/39f97b21499f862848df44f6f24f680d/untitled445.ipynb) and let us know can we close this issue. Thanks

**Comment by alanhdu** [comment]
Oh, sorrry! Yes, it looks like support has been added.
